lib/src/platform/src/platform.dart

```diff
@@ -18,27 +18,29 @@ class Platform {
   static final StreamsChannel streamsChannel =
       StreamsChannel('io.ably.flutter.stream', codec);
 
-  /// Initializing ably on platform side by invoking `register` platform method.
-  /// Register will clear any stale instances on platform.
-  static Future? _initializer;
+  /// This field will reset to true when the state is reset. This allows us to
+  /// if an app has been restarted, or a hot restart (not hot reload) has
+  /// happened.
+  static bool _shouldResetAblyClients = true;
 
-  static Future _initialize() async {
-    if (_initializer == null) {
+  /// Clears instances on the Platform side
+  static Future<void> _resetOldAblyClients() async {
+    if (_shouldResetAblyClients) {
```

Comment:
I think this check should be done where method call is done

Reply:
This has been moved in a separate PR https://github.com/ably/ably-flutter/pull/235/files which does more refactoring 2fd960927973d2197aac957f8e3e172c20743657